msm/src/circuit_design/witness.rs

```diff
@@ -301,26 +308,29 @@ impl<
             lookups: vec![lookups_row],
             fixed_selectors,
             phantom_cix: PhantomData,
+            assert_mapper: Box::new(|x| x),
         }
     }
 
     /// Sets a fixed selector, the vector of length equal to the
     /// domain size (circuit height).
-    pub fn set_fixed_selector(&mut self, sel: CIx, sel_values: Vec<F>) {
+    pub fn set_fixed_selector_cix(&mut self, sel: CIx, sel_values: Vec<F>) {
         if let Column::FixedSelector(i) = sel.to_column() {
             self.fixed_selectors[i] = sel_values;
         } else {
             panic!("Tried to assign values to non-fixed-selector typed column {sel:?}");
         }
     }
 
-    /// Generates proof inputs, repacking/collecting internal witness builder state.
-    pub fn get_proof_inputs(
-        &self,
-        domain: EvaluationDomains<F>,
-        lookup_tables_data: BTreeMap<LT, Vec<F>>,
-    ) -> ProofInputs<N_COL, F, LT> {
+    /// Sets all fixed selectors directly. Each item in `selectors` is
+    /// a vector of `domain_size` length.
+    pub fn set_fixed_selectors(&mut self, selectors: Vec<Vec<F>>) {
+        self.fixed_selectors = selectors
+    }
+
+    pub fn get_relation_witness(&self, domain: EvaluationDomains<F>) -> Witness<N_COL, Vec<F>> {
         let domain_size: usize = domain.d1.size as usize;
+
         // Boxing to avoid stack overflow
         let mut witness: Box<Witness<N_COL, Vec<F>>> = Box::new(Witness {
             cols: Box::new(std::array::from_fn(|_| Vec::with_capacity(domain_size))),
```

Comment:
Nit: I would always add an assert at the end of the function checking that the size of the vector is *exactly* `domain_size`. `with_capacity` does not enforce that.